activesupport/lib/active_support/core_ext/module/redefine_method.rb

```diff
@@ -0,0 +1,38 @@
+class Module
+  # Marks the named method as intended to be redefined, if it exists.
+  # Suppresses the Ruby method redefinition warning. Prefer
+  # #redefine_method where possible.
+  def silence_redefinition_of_method(method)
+    if method_defined?(method) || private_method_defined?(method)
+      # This suppresses the "method redefined" warning; the self-alias
+      # looks odd, but means we don't need to generate a unique name
+      alias_method method, method
```

Comment:
The validity of this self-alias pattern is potentially an open question: https://bugs.ruby-lang.org/issues/13574

But if we have to switch to a generated-name alternative, that's fine too.